File: backend/src/main/java/com/prologiccreations/traderssolution/service/super_classes/config/EmployeeService.java
package com.prologiccreations.traderssolution.service.super_classes.config;

import com.prologiccreations.traderssolution.dto.Response;
import com.prologiccreations.traderssolution.dto.config.ManagerDto;
import com.prologiccreations.traderssolution.model.config.Employee;
import com.prologiccreations.traderssolution.service.super_classes.CrudService;

import java.util.List;

public interface EmployeeService extends CrudService<Employee, Long> {

    Response<List<ManagerDto>> findAllManagers();

}

File: frontend/src/app/services/config/employee.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Manager } from 'src/app/model/config/manager.model';
import { AppResponse } from '../../dto/response.dto';
import { ConfigService } from './config.service';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class EmployeeServiceService {


  constructor(private http: HttpClient, private config: ConfigService) { }

  getManagers(): Observable<AppResponse<Manager[]>> {
    const url = `${this.config.apiUrl}/employee/managers`;
    return this.http.get<AppResponse<Manager[]>>(url);
  }
}
